dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java

```
@@ -1808,4 +1929,44 @@ private List<Resource> queryResourceList(Integer userId, int perm) {
     private AuthorizationType checkResourceType(ResourceType type) {
         return type.equals(ResourceType.FILE) ? AuthorizationType.RESOURCE_FILE_ID : AuthorizationType.UDF_FILE;
     }
+
+    /**
+     * check permission by comparing login user's tenantCode with tenantCode in the request
+     *
+     * @param loginUser user who currently logs in
+     * @param resTenantCode tenantCode in the request field "resTenantCode", can be different from the login user in the case of admin users.
+     * @param result  result Object containing different cases
+     * @return tenantCode
+     */
+    private String getTenantCodeIfuserValid(User loginUser, String resTenantCode, Result<Object> result) {
```

Please don't use the `Result<Object> result` as input param, we should avoid to change the input param.
You can throw the exception like `throw new ServiceException(Status.USER_NOT_EXIST)` when some check error.